File: 3475-minimum-operations-to-make-binary-array-elements-equal-to-one-i/3475-minimum-operations-to-make-binary-array-elements-equal-to-one-i.go
func minOperations(nums []int) int {
    cnt := 0
    n := len(nums)
    
    for i := 0; i < n - 2; i++ {
        if (nums[i] == 1) {
            continue
        }
        nums[i] ^= 1
        nums[i + 1] ^= 1
        nums[i + 2] ^= 1
        cnt++
    }

    if nums[n-2] == 1 && nums[n-1] == 1 {
        return cnt
    }
    return -1
}

File: 3475-minimum-operations-to-make-binary-array-elements-equal-to-one-i/3475-minimum-operations-to-make-binary-array-elements-equal-to-one-i.rs
impl Solution {
    pub fn min_operations(mut nums: Vec<i32>) -> i32 {
        let mut cnt = 0;
        let n = nums.len();

        for i in 0..n-2 {
            if nums[i] == 0 {
                nums[i] ^= 1;
                nums[i+1] ^= 1;
                nums[i+2] ^= 1;
                cnt+=1;
            }
        }

        if nums[n-2] == 1 && nums[n-1] == 1 {
            cnt
        } else {
            -1
        }
    }
}